### DISCUSSION TITLE
Subscription / Topic based metrics

### DISCUSSION THREAD

**Discussion by mudasirmirza** [emqx/emqx#13472] [Q&A] [answered]
Hi Everyone,

I was wondering if it is possible to know stats for each topic and subscription. Like I would like to know how many messages were published to each topic and to each subscription. 
Also it would be great to know how many messages were published by each client or subscriber for reliability and also great if we run a multi-tenant broker.

Thanks.

**Comment by id** [comment, marked as answer]
Hi @mudasirmirza.

To configure topic metrics please refer to https://docs.emqx.com/en/emqx/latest/observability/topic-metrics.html.

Metrics per client available on client page on dashboard, as well as via API https://docs.emqx.com/en/emqx/v5.7/admin/api-docs.html#tag/Clients/paths/~1clients~1%7Bclientid%7D/get.
